datafusion/core/src/physical_optimizer/repartition.rs

```diff
@@ -161,34 +161,14 @@ fn optimize_partitions(
         // leaf node - don't replace children
         plan
     } else {
-        let can_reorder_children =
-            match (plan.relies_on_input_order(), plan.maintains_input_order()) {
```

Comment:
I believe this is the last use of `maintains_input_order` -- so after this PR is merged, I will try and create a PR that removes this from the trait -- it is redundant with the `plan.output_order` I think